WorkflowUI/Sources/ViewControllerDescription/UIViewController+Extensions.swift

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2020 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if canImport(UIKit)
+
+    import UIKit
+
+    extension UpdateChildScreenViewController where Self: UIViewController {
+        /// Updates the view controller at the given `child` key path with the
+        /// `ViewControllerDescription`from `screen`. If the type of the underlying view
+        /// controller changes between update passes, this method will remove
+        /// the old view controller, create a new one, update it, and insert it into the view controller hierarchy.
+        ///
+        /// The view controller at `child` must be a child of `self`.
+        ///
+        /// - Parameters:
+        /// - parameter child: The `KeyPath` which describes what view controller to update. This view controller must be a direct child of `self`.
+        /// - parameter screen: The `Screen` instance to apply to the view controller.
+        /// - parameter environment: The `environment` to used when updating the view controller.
+        /// - parameter onChange: A callback called if the view controller instance changed.
+        ///
+        public func update<VC: UIViewController, ScreenType: Screen>(
+            child: ReferenceWritableKeyPath<Self, VC>,
+            with screen: ScreenType,
+            in environment: ViewEnvironment,
+            onChange: (VC) -> Void = { _ in }
+        ) {
+            let description = screen.viewControllerDescription(environment: environment)
+
+            let existing = self[keyPath: child]
+
+            if description.canUpdate(viewController: existing) {
+                // Easy path: Just update the existing view controller if we can do that.
+                description.update(viewController: existing)
```

Comment:
Do we potentially need to add subviews etc. here if they're not already added (due to the `isViewLoaded` checks below)?

Reply:
No – the assumption is the implementation of the view controller will do this in `loadView/viewDidLoad`, as custom parent VCs are expected to do.